### PR TITLE
SONARJAVA-3124 S3077: ignore some standard immutable types

### PR DESCRIPTION
**Pull request by quentin-jaquier-sonarsource**

### CODE REVIEW

File: java-checks/src/main/java/org/sonar/java/checks/VolatileNonPrimitiveFieldCheck.java [hunk 1]
```diff
@@ -36,6 +36,44 @@
...
+    "java.time.OffsetDateTime",
+    "java.time.OffsetTime",
+    "java.time.Year",
+    "java.time.YearMonth",
+    "java.time.ZoneId",
+    "java.time.ZoneOffset",
+    "java.time.ZonedDateTime",
+    "java.time.Duration",
+    "java.time.Period",
+    "java.util.Date",
```
Comment (saberduck) [inline java-checks/src/main/java/org/sonar/java/checks/VolatileNonPrimitiveFieldCheck.java]: `java.util.Date` is not immutable, see https://docs.oracle.com/javase/7/docs/api/java/util/Date.html#setDate(int)

File: java-checks/src/main/java/org/sonar/java/checks/VolatileNonPrimitiveFieldCheck.java [hunk 2]
```diff
@@ -36,6 +36,44 @@
   private static final String REF_MESSAGE = "Remove the \"volatile\" keyword from this field.";
   private static final String ARRAY_MESSAGE = "Use an \"Atomic%sArray\" instead.";
 
+  private static final List<String> STANDARD_IMMUTABLE_TYPES = Arrays.asList(
```
Comment (saberduck) [inline java-checks/src/main/java/org/sonar/java/checks/VolatileNonPrimitiveFieldCheck.java]: can you add `java.net.Inet4Address`, `java.net.Inet6Address`

File: java-checks/src/test/files/checks/VolatileNonPrimitiveFieldCheck.java [hunk 3]
```diff
@@ -1,3 +1,41 @@
+import java.util.concurrent.atomic.AtomicIntegerArray;
+import java.awt.Color;
+import java.awt.Cursor;
+import java.awt.Font;
+import java.io.File;
+import java.lang.Boolean;
```
Comment (saberduck) [inline java-checks/src/test/files/checks/VolatileNonPrimitiveFieldCheck.java]: no need to import `java.lang` , they are in scope by default

File: java-checks/src/main/java/org/sonar/java/checks/VolatileNonPrimitiveFieldCheck.java (hunk 1 above)
Comment (quentin-jaquier-sonarsource) [reply java-checks/src/main/java/org/sonar/java/checks/VolatileNonPrimitiveFieldCheck.java]: Indeed, nice catch, I have updated the ticket aswell!

**Pull request merged**